Narrow or remove the hole that pantsd punches in the ignore patterns for `.pids`

In #9636, we punched a hole in the ignore patterns in order to allow `pantsd` to watch its own pidfile. But in cases where users had previously used `pantsd`, this resulted in a failure to deal with an orphaned `watchman` socket file (`watchman` is now disabled by default, so this only impacts users who) which had previous been ignored.

```
22:32:58:201 [ERROR] 1 Exception encountered:
  Exception: Failed to scan directory "$workspace/.pids/watchman": Expected File, Dir or Link, but ".pids/watchman/watchman.sock" (relative to "$workspace") was a FileType(FileType { mode: 49152 })
```

Narrowing the pattern to just the `.pids/pantsd` subdirectory is challenging (possibly not an option) because gitignore patterns do not allow re-including the children (`.pids/pantsd` in this case) of ignored parents (`.pids`, due to `.*`).

--- a/src/python/pants/init/options_initializer.py
+++ b/src/python/pants/init/options_initializer.py
@@ -118,9 +118,12 @@ class OptionsInitializer:
 
         add(global_options.pants_workdir)
         add(global_options.pants_distdir)
-        # NB: We punch a hole in the ignore patterns to allow pants to directly watch process
-        # metadata that is written to disk.
+        # TODO: We punch a hole in the ignore patterns to allow pantsd to directly watch process
+        # metadata that is written to disk, but we re-ignore the watchman directory (which
+        # contains a named pipe). Over time, as more of the pantsd server components are ported to
+        # rust, we will be able to remove this special case.
         add(global_options.pants_subprocessdir, include=True)
+        add(os.path.join(global_options.pants_subprocessdir, "watchman"))
 
         return pants_ignore
 

